packages/flask/src/UIAnimatedBalance/UIAnimatedBalance.tsx

```diff
@@ -0,0 +1,286 @@
+                },
+            ],
+        };
+    });
+
+    return (
+        <AnimatedUIImage
+            source={source}
+            resizeMode={'contain'}
+            style={[iconStaticStyle, iconAnimatedStyle]}
+        />
+    );
+}
+
+export function UIAnimatedBalance({
+    integerVariant = TypographyVariants.TitleLarge,
+    fractionalVariant = TypographyVariants.LightLarge,
+    value,
+    maxFractionalDigits = 2,
+    icon,
+    loading = false,
+}: {
+    value: number;
+    icon: UIImageProps['source'];
+    integerVariant?: TypographyVariants;
+    fractionalVariant?: TypographyVariants;
+    maxFractionalDigits?: number;
+    loading?: boolean;
+}) {
+    const theme = useTheme();
+    const valueHolder = useSharedValue(value);
+    const { decimalSeparator } = uiLocalized;
+
+    React.useEffect(() => {
+        if (valueHolder.value === value) {
+            return;
+        }
+
+        valueHolder.value = withTiming(value, {
+            duration: 400,
+            easing: Easing.inOut(Easing.ease),
+        });
+    }, [value]);
+
+    const formatted = useDerivedValue(() => {
+        const integer = Math.floor(valueHolder.value);
+        const fractional = Math.floor((valueHolder.value - integer) * 10 ** maxFractionalDigits);
+        return {
+            integer: integer.toString(),
+            fractional: fractional.toString().padEnd(maxFractionalDigits, '0'),
+        };
+    });
+
+    /**
+     * (savelichalex):
+     * In order to understand why we need it at all,
+     * you should understand that numeric letters
+     * actually have different width, so when we change
+     * the value of a balance fast, the whole width of it
+     * also will have different width, and it will be changing fast.
+     * Since icon is positioned after balance, it will cause
+     * change of position for icon, that looks like icon is shaking.
+     * So to prevent that I drew balance two times:
+     * 1. Visible part, that draw actual balance
+     * 2. Hidden part, it's a copy of the visible balance with
+     *    all numerical letters replaced to "wider" ones
+     * Example:
+     * 1234.67 -> 5555.55
+     *
+     * We render icon with hidden part, as it changes width
+     * less frequently then visible one, therefore it looks stable.
+     * Visible part is just placed above with absolute positioning.
+     */
+    const normalized = useSharedValue(value.toString());
+    useAnimatedReaction(
+        () => {
+            return formatted.value;
+        },
+        (state, prevState) => {
+            // To change the hidden input only in case the length is changed
+            if (
+                prevState != null &&
+                state.integer.length + state.fractional.length ===
+                    prevState.integer.length + prevState.fractional.length
+            ) {
+                return;
+            }
+
+            const numRegExp = /[0-9]/g;
+            const wideNumLiteral = '5';
```

Comment:
The constants should probably be taken out?

Reply:
I don't want them to be captured and turned into SharedValues :) especially RegExp, it works badly with reanimated when it's captured

Reply:
It won't properly work with the tag `@inline`?

Reply:
I did only numbers for now :)

Reply:
it's a pity, but not critical :)